src/lib/QueryType.hs

```diff
@@ -236,9 +240,9 @@ sourceNameType v = do
       UDataConVar v' -> lookupEnv v' >>= \case DataConBinding _ _ e -> getType e
       UClassVar   v' -> lookupEnv v' >>= \case ClassBinding  def -> return $ getClassTy def
       UMethodVar  v' -> lookupEnv v' >>= \case MethodBinding _ _ e  -> getType e
-      UEffectVar   _ -> error "not implemented: sourceNameType::UEffectVar"
-      UEffectOpVar _ -> error "not implemented: sourceNameType::UEffectOpVar"
-      UHandlerVar  _ -> error "not implemented: sourceNameType::UHandlerVar"
+      UEffectVar  v' -> getType $ userEffect v'
```

Comment:
This rule is a little surprising to me. As far as I understand the rhs of `->` always evaluated to `EffectRowKind`. But the var that names an effect is not an effect row, it's an effect. We might need to add another kind in primitives to express that

Reply:
Maybe I'm confused about the context of this code? I thought these names would appear only in expressions like `MyEff` in `{MyEff|effs}`. I need a Type here but the only constructor that can take a user effect is `Eff`, which expects an effect row. So I construct just `{MyEff|}` as a singleton effect row. I didn't see anything obviously wrong with this, but maybe there is and so we _do_ need another kind?